csstools/postcss-plugins
# postcss tape crafted for this repo

- cli -> lib
- simplified API
- pretty diff
- github annotations
- test multiple PostCSS versions
- unify PostCSS versions used for tests (this removes `postcss` from plugin dev dependencies)
- verify that the result is still parsable with PostCSS
- verify source maps (no `<no-source>`)
- verify both `commonjs` and `es modules`
- verify [plugin standards](https://github.com/postcss/postcss/blob/main/docs/guidelines/plugin.md#54-include-postcss-plugin-keyword-in-packagejson)

-----

Making this a lib instead of a CLI simplifies a lot of things.
As this is a private package we know that we only use this in ways where a lib is possible.

-----

In general this change trades API complexity for test complexity.

1. The public interface is much smaller and simpler.
2. There are more tests and checks done with the same information.

-----

The tests for the package use `bash` and `git` to avoid adding real code that would itself also need tests.

- outputs on `stdout` and `stderr` for a test run are piped to one file
- exit code for a test run is piped to another file

These are compared with `git diff` to check if the new PostCSS Tape handles each case as expected.

-----

I have only updated a couple of plugins to use the new test suite.
I do not want to make review of this package harder by having too many changes at once.
A follow-up PR will migrate all other plugins.

If I maybe missed some part of the original PostCSS Tape that is used by a plugin, please let me know.

We could also open up select CI checks to forks.
This combined with the improved feedback will make it easier for contributors to iterate without needing to wait for maintainer feedback.
